credoai/evaluators/deepchecks.py

```diff
@@ -0,0 +1,143 @@
+    for more details.
+
+    This evaluator provides some redundant functionality. For instance, metrics which can be
+    calculated using the Performance evaluator can potentially be calculated by deepchecks
+    (and thus this evaluator) as well. The same applies to the FeatureDrift evaluator.
+    When a choice exists, the best practice dictates that the "Lens native" evaluator should
+    be used in preference to deepchecks, since output formats of other evaluators is generally
+    consistent, while this deepchecks evaluator outputs results in a highly structured JSON format.
+
+
+    Parameters
+    ----------
+    name : str, optional
+        Name of the supplied deepchecks suite
+    checks : List-like, optional
+        A list of instantiated deepchecks checks objects (e.g. BoostingOverfit, CalibrationScore)
+        #TODO allow list of strings?
+    """
+
+    name = "Deepchecks"  ##This is going to go away once we merge with Ian's PR #214
+
+    required_artifacts = {"model", "assessment_data", "training_data"}
+    # all artifacts are OPTIONAL; All that's required is that at least one of these is
+    # provided. The evaluator's custom validation function checks for this.
+
+    def __init__(
+        self,
+        suite_name: Optional[str] = "Credo_Deepchecks_Suite",
+        checks: Optional[List[BaseCheck]] = DEFAULT_CHECKS,
+    ):
+        super().__init__()
+        self.name = suite_name
+        self.checks = checks
+
+    def _setup(self):
+        # Set artifacts
+
+        # All artifacts are optional and thus any could be NoneType
+        # Internal (lens) validation ensures that at least one artifact is valid
+        self.model = self.model
+        self.test_dataset = self.assessment_data
+        self.train_dataset = self.training_data
+
+    def _setup_deepchecks(self):
+        if self.test_dataset:
+            self.test_dataset = Dataset(
+                df=self.test_dataset.X, label=self.test_dataset.y
+            )
+
+        if self.train_dataset:
+            self.train_dataset = Dataset(
+                df=self.train_dataset.X, label=self.train_dataset.y
+            )
+
+        if self.model:
+            self.deepchecks_model = self.model.model_like
+
+        self.suite = Suite(name=self.name)
+        for check in self.checks:
+            self.suite.add(check)
+            # doing this as a for-loop list seems to be the only way
+            # deepchecks won't let you pass a whole list of checks, which is...silly?
+
+    def evaluate(self):
+        """
+        Execute any data/model processing required for the evaluator.
+
+        Populates the self.results object.
+
+        Returns
+        -------
+        self
+        """
+        self._setup_deepchecks()
+
+        if self.train_dataset and self.test_dataset:
+            self.results = [
+                DeepchecksContainer(
+                    self.name,
+                    self.suite.run(
```

Comment:
Though it is simple to write `self.suite.run`, there's slightly more in `evaluate` than I'd like to see. A separate function called `run_suite` would be better, and then evaluate takes the output of that and just wraps it into a `DeepchecksContainer`